File: 0055-jump-game/0055-jump-game.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
    let n = nums.length;
    let far = 0;
    for(let i = 0; i < n; ++i) {
        far = Math.max(far, i + nums[i]);
        if(i >= far && i != n - 1) return false;
        if(far >= n - 1) return true;
    }
    return true;
};

File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        if n == 1: return True
        far = 0
        for i in range(n):
            far = max(far, i + nums[i])
            if i >= far and i != n - 1: return False;
            if far >= n - 1: return True
        
        return True